Automattic-Tracks-iOS/Crash Logging/CrashLogging.swift

```diff
@@ -278,6 +286,104 @@ extension CrashLogging {
+    /// This function is based on the original Sentry implementation for React native: https://github.com/getsentry/sentry-react-native/blob/aa4eb11415cbb73bbd0033e4f0926b539d22315b/ios/RNSentry.m#L118-L158
+    ///
+    /// - Parameters:
+    ///   - envelopeDict: The envelope object.
+    public func logEnvelope(_ envelopeDict: [String: Any]) {
+        if JSONSerialization.isValidJSONObject(envelopeDict) {
+            guard let headerDict = envelopeDict["header"] as? [String: Any] else {
+                DDLogError("⛔️ Unable to send envelope to Sentry – header is not defined in the envelope.")
+                return
+            }
+            guard let headerEventId = headerDict["event_id"] as? String else {
+                DDLogError("⛔️ Unable to send envelope to Sentry – event id is not defined in the envelope header.")
+                return
+            }
+            guard let payloadDict = envelopeDict["payload"] as? [String: Any] else {
+                DDLogError("⛔️ Unable to send envelope to Sentry – payload is not defined in the envelope.")
+                return
+            }
+            guard let eventLevel = payloadDict["level"] as? String else {
+                DDLogError("⛔️ Unable to send envelope to Sentry – level is not defined in the envelope payload.")
+                return
+            }
+
+            // Define the envelope header
+            let sdkInfo = SentrySdkInfo.init(dict: headerDict)
+            let eventId = SentryId.init(uuidString: headerEventId)
+            let envelopeHeader = SentryEnvelopeHeader.init(id: eventId, andSdkInfo: sdkInfo)
+
+            guard let envelopeItemData = try? JSONSerialization.data(withJSONObject: payloadDict) else {
+                DDLogError("⛔️ Unable to send envelope to Sentry – payload could not be serialized.")
+                return
+            }
+
+            let itemType = payloadDict["type"] as? String ?? "event"
+            let envelopeItemHeader = SentryEnvelopeItemHeader.init(type: itemType, length: UInt(bitPattern: envelopeItemData.count))
+            let envelopeItem = SentryEnvelopeItem.init(header: envelopeItemHeader, data: envelopeItemData)
+            let envelope = SentryEnvelope.init(header: envelopeHeader, singleItem: envelopeItem)
+
+            #if DEBUG
+            SentrySDK.currentHub().getClient()?.capture(envelope: envelope)
+            #else
+            if eventLevel == "fatal" {
+                // Storing to disk happens asynchronously with captureEnvelope
+                SentrySDK.currentHub().getClient()?.store(envelope)
+            } else {
+                SentrySDK.currentHub().getClient()?.capture(envelope: envelope)
+            }
+            #endif
+        } else {
+            DDLogError("⛔️ Unable to send envelope to Sentry – envelope is not a valid JSON object.")
+        }
+    }
+}
```

Comment:
Hybrid SDKs send the events as [envelopes](https://develop.sentry.dev/sdk/envelopes/), it's basically an object that contains all the data required to ingest an event into Sentry.

This implementation is based on the [original version of the Sentry React native SDK (version `2.4.2`)](https://github.com/getsentry/sentry-react-native/blob/a1b314781fb673becfb572a97a297b9bb8e7a849/ios/RNSentry.m#L130-L172). I'd like to note that the code reference is from version `2.4.2` because is the last version that [uses Sentry iOS SDK version `6.x`](https://github.com/getsentry/sentry-react-native/blob/a1b314781fb673becfb572a97a297b9bb8e7a849/RNSentry.podspec#L20) that should match the version we use for crash logging ([currently `6.x` from the podspec](https://github.com/Automattic/Automattic-Tracks-iOS/blob/36db74cceaf98e107e0c713eaae81eb4139c2059/Automattic-Tracks-iOS.podspec#L40)).